src/app/zap-templates/templates/app/MatterIDL.zapt

```diff
@@ -0,0 +1,80 @@
+// This IDL was generated automatically by ZAP.
+// It is for view/code review purposes only.
+
+{{#chip_shared_structs}}
+{{#if struct_is_fabric_scoped}}
+[fabric_scoped_by=k{{asUpperCamelCase struct_fabric_idx_field}}]
+{{/if}}
+struct {{name}} {
+  {{#zcl_struct_items}}
+  {{type}} k{{asUpperCamelCase label}} = {{fieldIdentifier}};
+  {{/zcl_struct_items}}
+}
+
+{{/chip_shared_structs}}
+{{!TODO: consider #chip_clusters as iteration point as well.
+{{!      Unsure about the differences}}
+{{#all_user_clusters}}
+{{side}} cluster {{asUpperCamelCase name}} = {{code}} {
+  {{#zcl_enums}}
+  enum {{asUpperCamelCase name}} : {{type}} {
+    {{#zcl_enum_items}}
+    k{{asUpperCamelCase label}} = {{value}};
+    {{/zcl_enum_items}}
+  }
+
+  {{/zcl_enums}}
+  {{#chip_cluster_specific_structs}}
+  {{#if struct_is_fabric_scoped}}
+  [fabric_scoped_by=k{{asUpperCamelCase struct_fabric_idx_field}}]
+  {{/if}}
+  struct {{name}} {
+    {{#zcl_struct_items}}
+    {{type}} k{{asUpperCamelCase label}} = {{fieldIdentifier}};
```

Comment:
Worth indicating when the field is optional.

Reply:
Updated with support for list/nullable/optional.